docs/content/contrib-adding-builtin-functions.md

```diff
@@ -0,0 +1,156 @@
+---
```

Comment:
I just realized there is more in-depth documentation for built-in:s at `extensions/#custom-built-in-functions-in-go`. Should probably link to there from here.

Reply:
That is a little different, though: it's about extending OPA, so you'd build your own binary with the extra built-in func included. This document describes how to contribute to OPA, so the new built-in function would become part of "core OPA" so to speak. There certainly is overlap, but it's a different twist.

Reply:
It is different for sure, but I think @johanfylling is right that it would be good to reference here.. at least in the sense that one step that's currently not mentioned is to first have some form of approval for the built-in. Without that—i.e. for someone who wants to integrate Kafka/Redis/DynamoDB/Whatever—the custom function is probably a better alternative.

Reply:
On a related note, this feels a bit like these are two changes: a docs addition and new built-in functions.

It would be nice if we untangled that a bit, we could massage it into two commits -- two PRs would be another option, but since it's here already, it's OK I think.

Reply:
Yea, I agree; I should have gone that route.